### DISCUSSION TITLE
ImageLoader for composable's Preview

### DISCUSSION THREAD

**Discussion by miguelaboliveira** [coil-kt/coil#2761] [Q&A] [answered]
Hi there!
I went through the documentation and could not find much help for this scenario.

I'm using coil3 in context of a library, which means that I must not use the singleton `ImageLoader` (using -core variants of coil). As such, I should pass an instance of the loader through my composables hierarchy up to some `AsyncImage`.
Eventually I must have some kind of "fake" loader to be used in those composables Preview. What is the best way for me to have that?

Also, somewhat related, should I "remember" the instance of the loader?

Thank you!

**Comment by colinrtwhite** [comment, marked as answer]
[More info on previews](https://coil-kt.github.io/coil/compose/#previews).

> Also, somewhat related, should I "remember" the instance of the loader?

Yes, you'll want to create one `ImageLoader` and share it around. This is documented [here](https://coil-kt.github.io/coil/image_loaders/).
